build.d

```diff
@@ -0,0 +1,756 @@
+        writeln("  --download-imgui         Download and build imgui");
+        writeln("  --download-sokol-tools   Download and setup sokol-tools");
+        return;
+    }
+    else
+    {
+        writeln("Currently configuration:");
+        writeln("Using compiler: ", compiler);
+        writeln("Using target: ", target);
+        writeln("Using optimize: ", optimize);
+        writeln("Using backend: ", sokol_backend);
+        writeln("Get imgui-libs: ", downloadIMGUI);
+        writeln("Get sokol-tools: ", downloadSHDC);
+        writeln("Get EmSDK: ", downloadEmsdk);
+        writeln("Get Zig: ", downloadZig);
+    }
+
+    LibSokolOptions options = {
+        target: target,
+        optimize: optimize,
+        toolchain: compiler,
+        backend: sokol_backend,
+        use_wayland: opt_use_wayland,
+        use_x11: opt_use_x11,
+        use_egl: opt_use_egl,
+        with_sokol_imgui: opt_with_sokol_imgui,
+    };
+
+    // Download and setup Emscripten SDK if requested
+    if (downloadEmsdk)
+        getEmSDK;
+    if (downloadIMGUI)
+        getIMGUI;
+    if (downloadZig)
+        getZigToolchain;
+    if (downloadSHDC)
+        buildShaders;
+
+    // build libsokol
+    buildLibSokol(options);
+}
+
+void getZigToolchain() @safe
+{
+    writeln("Downloading and setting up Zig toolchain...");
+    string rootpath = absolutePath(buildPath("vendor"));
+
+    version (Windows)
+        immutable string ext = ".zip", exe = ".exe";
+    else
+        immutable string ext = ".tar.xz", exe = "";
+
+    version (Windows)
+        immutable string os = "windows";
+    else version (Linux)
+        immutable string os = "linux";
+    else version (OSX)
+        immutable string os = "macos";
+    else version (FreeBSD)
+        immutable string os = "freebsd";
+
+    version (X86_64)
+        immutable string arch = "x86_64";
+    else version (X86)
+        immutable string arch = "x86";
+    else version (AArch64)
+        immutable string arch = "aarch64";
+
+    immutable string filename = fmt("zig-%s-%s-%s", os, arch, zig_version);
+    writeln("file: ", filename ~ ext);
+    writeln("rootpath: ", rootpath);
+
+    scope (exit)
+    {
+        if (exists(filename ~ ext))
+            remove(filename ~ ext);
+    }
+
+    // Check if the file already exists
+    if (!exists(rootpath))
+    {
+        download(fmt("https://ziglang.org/download/%s/%s", zig_version, filename ~ ext), filename ~ ext);
+
+        // Extract Zig toolchain
+        if (ext.endsWith("zip"))
+            extractZip(filename ~ ext, rootpath);
+        else
+            extractTarXZ(filename ~ ext, rootpath);
+    }
+
+    // move from vendor/zig-{os}-{arch}-{version} to vendor/zig
+    immutable string extractedPath = buildPath(rootpath, filename);
+    immutable string newPath = buildPath(rootpath, "zig");
+    if (exists(extractedPath) && !exists(newPath))
+        rename(extractedPath, newPath);
+
+    rootpath = newPath;
```

Comment:
commit: https://github.com/kassane/sokol-d/pull/70/commits/1f2d8f09fc2cf12cdab9b95ada08868319c2ffb3
Sadly, for windows rename folder get error (not found)!!

So the idea behind **Zig** as a compiler alternative to **C/C++**, to make it more like a `cc-rs`, or rather a `cargo-zigbuild`.

Reply:
It's uncomfortable to set up many compilers... worse msvc/msbuild! :disappointed: